cle/backends/__init__.py

```diff
@@ -176,15 +176,21 @@ def __init__(self, owner, name, addr, size, binding, sym_type, sh_info):
         self.sh_info = sh_info if sh_info != 'SHN_UNDEF' else None
         self.resolved = False
         self.resolvedby = None
+        self.is_static = is_static
         if (claripy and isinstance(self.addr, claripy.ast.Base)) or self.addr != 0:
             self.owner_obj.symbols_by_addr[self.addr] = self
             # would be nice if we could populate demangled_names here...
 
             #demangled = self.demangled_name
             #if demangled is not None:
             #    self.owner_obj.demangled_names[self.name] = demangled
+        if is_static:
+            self.resolve(self)
 
     def resolve(self, obj):
+        if self.is_static and obj is not self:
+            raise CLEError("Static symbols cannot be resolved by another symbol.")
+
         self.resolved = True
         self.resolvedby = obj
         self.owner_obj.resolved_imports.append(self)
```

Comment:
It's a list, at the moment. I'm not sure to what extent this matters, since this property is never actually used except to add to the list.